plugins/transfer/transfer.plugin.zsh

```diff
@@ -0,0 +1,20 @@
+function transfer() {
+  if [ $# -eq 0 ]; then
+    echo "No arguments specified. Usage:\necho transfer /tmp/test.md\ncat /tmp/test.md | transfer test.md"
+    return 1
+  fi
+
+  tmpfile=$( mktemp -t transferXXX )
```

Comment:
The temp file might not be necessary. `curl` writes the progress bar to the tty and not stdout (I think), so you could just capture the URL with `$(curl ... "https://transfer")`